oap-server/oal-rt/src/main/java/org/apache/skywalking/oal/rt/parser/AggregationFuncStmt.java

```diff
@@ -49,22 +50,23 @@ public ConditionExpression getNextFuncConditionExpression() {
         return funcConditionExpressions.get(funcConditionExpressionGetIdx++);
     }
 
-    public void addFuncArg(Argument argument) {
+    public void addFuncArg(Argument... argument) {
```

Comment:
An argument is an Argument Array. I think this is wrong from grammar level. We should discuss how to host an Array inside the `Argument`, or create a sub-type of `Argument`, or create an interface with two implementations??

Reply:
Yes ,  Do you  agree  with adding `arrayFuncParamExpression `  to `grammar` ?
The  demo  such as 
```
mq_transmission_latency = from(MessageQueueAccess.transmissionLatency).filter(operation == MessageQueueOperation.Consume).labeledLongAvg([topic,queue,queue,"1"]);

```
Let's  talk  all here  .

Reply:
Please explain this `topic,queue,queue,"1"` first. Is this meaning, first three elements of the array are from source attributes, and the last is a literal string?

Reply:
queue , topic is from  source  attributes , and  "1" is demo  just  for demonstrating  literal string

Reply:
Then, you still don't have funcParamExpression. Could you confirm whether you need this?

About the array type, do we need to make it type-sensitive? Such as building StringArray, rather than a general Array?

Reply:
>  you still don't have funcParamExpression

Yes , I would  confirm it .

> About the array type

Actually , I want to build the type of array  based on the parameter type , which will support base-type , such as  StringArray  , intArray，doubleArray， etc

Reply:
But, there is an issue about this. `SourceFrom` should be for `getting field value from a source's attribute`, and `Arg` should be a kind of literal declared from the OAL.

Now, my question would be, can't we build a field in the source including key-value pairs directly? And use `SourceFrom` rather than adding this complexity for the OAL? We are not really building an array mixed with attribute values and literal strings, right?

Reply:
I commented another concern https://github.com/apache/skywalking/issues/9676#issuecomment-1283735427 about the original proposal.